RedacteurPortaal.Api/Controllers/ArchiveController.cs

```diff
@@ -0,0 +1,240 @@
+        return this.Ok(dto.MediaVideoItems);
+    }
+
+    [HttpGet]
+    [Route("{archiveId}/AudioItems")]
+    public async Task<ActionResult<List<MediaAudioItem>>> GetAudioItems([FromRoute] Guid archiveId)
+    {
+        var archive = await this.grainService.GetGrain(archiveId);
+        var response = await archive.Get();
+        var dto = response.Adapt<ArchiveModel>();
+        return this.Ok(dto.MediaAudioItems);
+    }
+
+    [HttpGet]
+    [Route("{archiveId}/PhotoItems")]
+    public async Task<ActionResult<List<MediaPhotoItemDto>>> GetPhotoItems([FromRoute] Guid archiveId)
+    {
+        var archive = await this.grainService.GetGrain(archiveId);
+        var response = await archive.Get();
+        var dto = response.Adapt<ArchiveModel>();
+        return this.Ok(dto.MediaPhotoItems);
+    }
+
+    [HttpGet]
+    [Route("{archiveId}/Stories")]
+    public async Task<ActionResult<List<NewsItemDto>>> GetNewsItems([FromRoute] Guid archiveId)
+    {
+        var archive = await this.grainService.GetGrain(archiveId);
+        var response = await archive.Get();
+        var dto = response.Adapt<ArchiveModel>();
+        return this.Ok(dto.NewsItems);
+    }
+
+    [HttpGet]
+    [Route("{archiveId}/VideoItems/{videoItemGuid}")]
+    public async Task<IActionResult> GetVideoItem([FromRoute] Guid archiveId, [FromRoute] Guid videoItemGuid)
+    {
+        var grain = await this.grainService.GetGrain(archiveId);
+        var response = await grain.GetVideoItem(videoItemGuid);
+        var dto = response.Adapt<MediaVideoItem>();
+        return this.Ok(dto);
+    }
+
+    [HttpGet]
+    [Route("{archiveId}/AudioItems/{audioItemGuid}")]
+    public async Task<IActionResult> GetAudioItem([FromRoute] Guid archiveId, [FromRoute] Guid audioItemGuid)
+    {
+        var grain = await this.grainService.GetGrain(archiveId);
+        var response = await grain.GetAudioItem(audioItemGuid);
+        var dto = response.Adapt<MediaAudioItem>();
+        return this.Ok(dto);
+    }
+
+    [HttpGet]
+    [Route("{archiveId}/PhotoItems/{photoItemGuid}")]
+    public async Task<IActionResult> GetPhotoItem([FromRoute] Guid archiveId, [FromRoute] Guid photoItemGuid)
+    {
+        var grain = await this.grainService.GetGrain(archiveId);
+        var response = await grain.GetPhotoItem(photoItemGuid);
+        var dto = response.Adapt<MediaPhotoItem>();
+        return this.Ok(dto);
+    }
+
+    [HttpGet]
+    [Route("{archiveId}/Stories/{newsItemGuid}")]
+    public async Task<IActionResult> GetNewsItem([FromRoute] Guid archiveId, [FromRoute] Guid newsItemGuid)
+    {
+        var grain = await this.grainService.GetGrain(archiveId);
+        var response = await grain.GetNewsItem(newsItemGuid);
+        var dto = response.Adapt<NewsItemModel>();
+        return this.Ok(dto);
+    }
+
+    [HttpPost]
+    [Route("{archiveId}/")]
+    public async Task<ActionResult<ArchiveModel>> CreateArchive([FromBody] ArchiveDto archiveDTO)
+    {
```

Comment:
waarom moet hier een archive id in de route? Dit is toch een POST voor aanmaken ?

Reply:
aangepast